test/tchannel_bridge.js

```diff
@@ -0,0 +1,122 @@
+
+import _ from 'lodash';
+import {assert} from 'chai';
+import * as constants from '../src/constants';
+import ConstSampler from '../src/samplers/const_sampler';
+import path from 'path';
+import InMemoryReporter from '../src/reporters/in_memory_reporter';
+import opentracing from 'opentracing';
+import Tracer from '../src/tracer';
+import TChannel from 'tchannel';
+import TChannelBridge from '../src/tchannel_bridge.js';
+import TChannelAsThrift from 'tchannel/as/thrift';
+import TChannelAsJSON from 'tchannel/as/json';
+
+describe ('test tchannel span bridge', () => {
+    // BIG_TIMEOUT is useful for debugging purposes.
+    let BIG_TIMEOUT = 15000000;
+    let tracer = new Tracer(
+        'test-service',
+        new InMemoryReporter(),
+        new ConstSampler(true)
+    );
+    let bridge = new TChannelBridge(tracer);
+
+    let options = [
+        { description: 'tchannelAsJson', channelEncoding: TChannelAsJSON },
+        { description: 'tchannelAsThrift', channelEncoding: TChannelAsThrift }
+    ];
+
+    _.each(options, (o) => {
+
+        it (o.description + ' spans propagate through tchannel and preserve parent span properties', (done) => {
+            let originalSpan = tracer.startSpan('futurama');
+            originalSpan.setBaggageItem('leela', 'fry');
+            let contextForOutgoingCall = { span: originalSpan };
+
+            let server = new TChannel({
+                serviceName: 'server',
+                // force tracing on  in order to prove that overriding works
+                trace: true,
+                forceTrace: true
+            });
+            // Server calls client channel after it starts listening.
+            server.listen(4040, '127.0.0.1', onServerListening);
+
+            // Create the toplevel client channel.
+            let client = new TChannel({
+                // force tracing on  in order to prove that overriding works
+                trace: true,
+                forceTrace: true
+            });
+
+            // Create the client subchannel that makes requests.
+            let clientSubChannel = client.makeSubChannel({
+                serviceName: 'server',
+                peers: ['127.0.0.1:4040']
+            });
+
+            // Wrap the subchannel in an encoding
+            let encodedChannel = o.channelEncoding({
+                channel: clientSubChannel,
+                entryPoint: path.join(__dirname, 'thrift', 'echo.thrift') // ignored in json case
+            });
+
+            // register the server request function.
+            let context = {};
+            encodedChannel.register(server, 'Echo::echo', context, bridge.tracedHandler(handleServerReq));
+            function handleServerReq(context, req, head, body, callback) {
+                // assert that context is populated with new server span
+                assert.equal(originalSpan.context().traceIdStr, context.span.context().traceIdStr);
+
+                // assert that headers match the original span
```

Comment:
irrelevant. You can simply check that the new server span is a child of the client span, then these checked below are implied.